src/java/yaw/engine/items/HitBox.java

```diff
@@ -53,22 +51,219 @@ public boolean isIsCollisionWith(HitBox item){
         // test if edges of boundingBox1 intersect faces of boundingBox2
         for (int i = 0; i < tabIndexEdges.length; i += 2)
             for (int j = 0; j < tabIndexFaces.length; j += 4)
-                if (Collision.isIntersectSegmentAndFace(listVertexboundingBox1.get(tabIndexEdges[i]), listVertexboundingBox1.get(tabIndexEdges[i + 1]),
-                        listVertexboundingBox2.get(tabIndexFaces[j]), listVertexboundingBox2.get(tabIndexFaces[j + 1]),
-                        listVertexboundingBox2.get(tabIndexFaces[j + 2]), listVertexboundingBox2.get(tabIndexFaces[j + 3])))
+                if (isIntersectSegmentAndFace(listVertexHitBox1.get(tabIndexEdges[i]), listVertexHitBox1.get(tabIndexEdges[i + 1]),
+                        listVertexHitBox2.get(tabIndexFaces[j]), listVertexHitBox2.get(tabIndexFaces[j + 1]),
+                        listVertexHitBox2.get(tabIndexFaces[j + 2]), listVertexHitBox2.get(tabIndexFaces[j + 3])))
                     return true;
 
-        // XXX: the rest is superfluous ?
 
-        // test if edges of boundingBox2 intersect faces of boundingBox1
-        for (int i = 0; i < tabIndexEdges.length; i += 2)
-            for (int j = 0; j < tabIndexFaces.length; j += 4)
-                if (Collision.isIntersectSegmentAndFace(listVertexboundingBox2.get(tabIndexEdges[i]), listVertexboundingBox2.get(tabIndexEdges[i + 1]),
-                        listVertexboundingBox1.get(tabIndexFaces[j]), listVertexboundingBox1.get(tabIndexFaces[j + 1]),
-                        listVertexboundingBox1.get(tabIndexFaces[j + 2]), listVertexboundingBox1.get(tabIndexFaces[j + 3])))
-                    return true;
+        return false;
+    }
+
+    /**This method compute the max and min coordinates of the two hitboxes then check if the calling hitbox coordinates are bounded by the second one
+     *
+     * @param item the item with which the containment is check
+     * @return
+     */
+    public boolean isContainedIn(HitBox item){
```

Comment:
This is another concern ... the current PR should not change the intersection code